test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -96,30 +96,187 @@ public void DotnetRestore_WithAuthorSignedPackage_Succeeds()
                         new Dictionary<string, string>(),
                         attributes);
 
                     ProjectFileUtils.WriteXmlToFile(xml, stream);
                 }
 
+                var projectDir = Path.GetDirectoryName(workingDirectory);
+                //Directory.CreateDirectory(projectDir);
+                var configPath = Path.Combine(projectDir, "NuGet.Config");
+
+                //set nuget.config properties
+                var doc = new XDocument();
+                var configuration = new XElement(XName.Get("configuration"));
+                doc.Add(configuration);
+
+                var config = new XElement(XName.Get("config"));
+                configuration.Add(config);
+
+                var trustedSigners = new XElement(XName.Get("trustedSigners"));
+                configuration.Add(trustedSigners);
+
+                var signatureValidationMode = new XElement(XName.Get("add"));
+                signatureValidationMode.Add(new XAttribute(XName.Get("key"), "signatureValidationMode"));
+                signatureValidationMode.Add(new XAttribute(XName.Get("value"), "require"));
+                config.Add(signatureValidationMode);
+
+                //add trusted signers
+                var author = new XElement(XName.Get("author"));
+                author.Add(new XAttribute(XName.Get("name"), "microsoft"));
+                trustedSigners.Add(author);
+
+                var certificate = new XElement(XName.Get("certificate"));
+                certificate.Add(new XAttribute(XName.Get("fingerprint"), "3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE"));
+                certificate.Add(new XAttribute(XName.Get("hashAlgorithm"), "SHA256"));
+                certificate.Add(new XAttribute(XName.Get("allowUntrustedRoot"), "false"));
+                author.Add(certificate);
+
+                var repository = new XElement(XName.Get("repository"));
+                repository.Add(new XAttribute(XName.Get("name"), "nuget.org"));
+                repository.Add(new XAttribute(XName.Get("serviceIndex"), "https://api.nuget.org/v3/index.json"));
+                trustedSigners.Add(repository);
+
+                var rcertificate = new XElement(XName.Get("certificate"));
+                rcertificate.Add(new XAttribute(XName.Get("fingerprint"), "0E5F38F57DC1BCC806D8494F4F90FBCEDD988B46760709CBEEC6F4219AA6157D"));
+                rcertificate.Add(new XAttribute(XName.Get("hashAlgorithm"), "SHA256"));
+                rcertificate.Add(new XAttribute(XName.Get("allowUntrustedRoot"), "false"));
+                repository.Add(rcertificate);
+
+                var owners = new XElement(XName.Get("owners"));
+                owners.Add("dotnetframework;microsoft");
+                repository.Add(owners);
+
+                File.WriteAllText(configPath, doc.ToString());
+
                 _msbuildFixture.RestoreProject(workingDirectory, projectName, args: string.Empty);
             }
```

Comment:
Do we need assertion after running restore command?

Reply:
Assertions are performed [here](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs#L155-L168)

Reply:
Thanks for clarification.